src/containers/ActorsList/ActorsList.tsx

```diff
@@ -0,0 +1,183 @@
+import { Column } from "primereact/column";
+import {
+  useGetActorsQuery,
+  useUpdateActorMutation,
+  useDeleteActorMutation,
+} from "@services/actors";
+import { Button } from "primereact/button";
+import { Controller, useForm } from "react-hook-form";
+import { Actor } from "@utils/db";
+import { InputText } from "primereact/inputtext";
+import { MultiSelect, MultiSelectChangeEvent } from "primereact/multiselect";
+import { categoryTags } from "./categoryTags";
+
+export const ActorsList: FC = () => {
+  const [editingRow, setEditingRow] = useState<number | null>(null);
+  const { control, reset } = useForm<Actor>();
+  // I just found out that primereact does the sorting by itself
+  // but still will pass the sorting parameters for initial sorting.
+  const { data: actors, isLoading } = useGetActorsQuery({
+    sortBy: "n",
+    order: "asc",
+  });
+
+  const [updateActor] = useUpdateActorMutation();
+  const [deleteActor] = useDeleteActorMutation();
+
+  const saveChanges = useCallback(async () => {
+    await updateActor({
+      id: editingRow,
+      n: control._formValues["n"],
+      c: control._formValues["c"],
+    });
+    setEditingRow(null);
+    reset();
+  }, [setEditingRow, updateActor, editingRow, control, reset]);
+
+  const startEditing = useCallback(
+    (rowData: Actor) => {
+      reset();
+      setEditingRow(rowData.id);
+    },
+    [setEditingRow],
+  );
+
+  const deleteRow = useCallback(
+    async (actor: Actor) => {
+      deleteActor(actor);
+    },
+    [deleteActor],
+  );
+
+  const cancelEditing = useCallback(() => {
+    setEditingRow(null);
+    reset();
+  }, [setEditingRow, reset]);
+
+  const isEditing = useCallback(
+    (rowId: number) => editingRow === rowId,
+    [editingRow],
+  );
+
+  const nameEditor = useCallback(
+    (rowData: Actor) => (
+      <Controller
+        name="n"
+        control={control}
+        defaultValue={rowData.n}
+        render={({ field }) => (<InputText {...field} />)}
+      />
+    ),
+    [control],
+  );
+
+  const categoriesEditor = useCallback(
+    (rowData: Actor) => (
+      <Controller
+        name="c"
+        control={control}
+        defaultValue={rowData.c}
+        render={({ field }) => (
+          <MultiSelect
+            className={"max-w-[400px]"}
+            value={field.value}
+            optionValue={"value"}
+            optionLabel={"label"}
+            options={categoryTags}
+            onChange={(e: MultiSelectChangeEvent) => field.onChange(e.value)}
+            placeholder={"Select Categories"}
+            display={"chip"}
+          />
+        )}
+      />
+    ),
+    [control],
+  );
+
+  const actionBodyTemplate = useCallback(
+    (rowData: Actor) => {
+      return isEditing(rowData.id) ? (
+        <div className={"flex gap-x-2"}>
+          <Button
+            icon={"pi pi-times"}
+            rounded
+            outlined
+            severity={"danger"}
+            onClick={cancelEditing}
```

Comment:
Instead of setting “onClick” on each button I would better put “data-attr” on it, and listen for clicks in single handler that is set on the whole table. This will let us avoid creating excessive listeners.

I just was in rush yesterday and forgot to do it